crates/bevy_reflect/src/struct_trait.rs

```diff
@@ -508,6 +508,44 @@ impl Debug for DynamicStruct {
     }
 }
 
+impl FromIterator<(String, Box<dyn Reflect>)> for DynamicStruct {
```

Comment:
Nit: Is it possible to take `Into<Cow<'static, str>>` instead of just `String`?

Reply:
> Insert has the borrowed form to avoid creating a string when a field by the same name already exists, but that's not useful here because we're creating a new struct and need to store the string.

What do you mean? I was thinking that by taking `Into<Cow<'static, str>>` we could allow users to avoid writing `.to_string()` if they're passing string literals. `ToString` could also work but I think we want to plan for the future where we use a custom `CowStr` utility type.

> Though do you think it makes sense to issue a warning if the iterator contains the same name multiple times?

I don't think we have to. The insert methods don't give any warning. If it becomes a common stumbling block for users we can consider adding it in the future.